java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/model/GenericMqttMetadata.java

```diff
@@ -0,0 +1,66 @@
+package com.cumulocity.generic.mqtt.client.model;
+
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+
+/**
+ * Metadata for {@link GenericMqttMessage}.
+ */
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class GenericMqttMetadata {
+
+    /**
+     * The client id of the MQTT client.
+     */
+    private String clientId;
+
+    /**
+     * The message id of the message.
+     */
+    private int messageId;
+
+    /**
+     * The duplicate delivery flag.
+     * <p>
```

Comment:
suggestion: use @return for this. For example: @return true if this is a resent message, false otherwise'.

Or '..., false if this the first time this message has been sent to the receiver.'

Reply:
DOne.